tests/python_package_test/test_engine.py

```diff
@@ -1545,6 +1545,35 @@ def test_refit():
     assert err_pred > new_err_pred
 
 
+def test_refit_dataset_params():
+    # check refit accepts dataset_params
+    X, y = load_breast_cancer(return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    lgb_train = lgb.Dataset(X_train, y_train)
+    train_params = {
+        'objective': 'binary',
+        'metric': 'binary_logloss',
+        'verbose': -1,
+        'min_data': 10,
```

Comment:
```suggestion
```

I was confused by the appearance of `min_data` in the parameters passed to `lgb.train()`, as it's only used in `lgb.Dataset`: https://lightgbm.readthedocs.io/en/latest/Parameters.html#min_data_in_leaf

Please remove it here.